### DISCUSSION TITLE
Add reset_rotation_smoothing method to Camera2D

### DISCUSSION THREAD

**Discussion by farfind-gpt** [godotengine/godot-proposals#10056] [2D]
Hello everyone. While I was developing my game, I ran into a small problem: I needed to reset 2D camera rotation smoothing. I was looking for a method like the `reset_smoothing` but for rotation and didn't find it. After some time I found a solution:

```gdscript
camera.rotation_smoothing_enabled = false
await get_tree().process_frame
camera.rotation_smoothing_enabled = true
```

But I think it is better to have some method to do it directly, so [I tried to add that method](https://github.com/farfind-gpt/godot_reset_camera_smoothing/commit/769e9587e0dc4bb499752f9e26fe801d00dec7a1). And it seems like it works.

I think it also may be useful for other developers, but I'm not sure. I'm also not sure I made it best way.

So should I open a PR to Godot repository? Is it a good implementation of my idea or should I do it other way? Is it even a good idea or will this method be useless for most developers?

Update: a [PR](https://github.com/godotengine/godot/pull/94349) was opened.

**Comment by Calinou** [comment]
Out of curiosity, what do you need to reset 2D camera rotation smoothing for (but not position at the same time)?

**Reply by farfind-gpt** [reply]
I need to reset both rotation and position smoothing. I can reset position smoothing using the `reset_smoothing` method, but, if I am not mistaken, there is no similar method for rotation smoothing. Is there some method to reset both position and rotation smoothing at the same time?

**Reply by Calinou** [reply]
This sounds like a bug. I'd expect `reset_smoothing()` to reset both position and rotation smoothing. One exception for this use case might be for origin shifting though (if your camera is rotating at the same time as the origin shifting happens).

**Reply by farfind-gpt** [reply]
Sorry, but I don't understand what an origin shifting is. Do you mean changing position of a camera parent node?


Anyway, I see three solutions:

1. We can make `reset_smoothing` reset both position and rotation smoothing.

2. We can make `reset_smoothing` reset both position and rotation smoothing but also add `reset_position_smoothing` and `reset_rotation_smoothing` methods.

3. We can add `reset_position_smoothing` and `reset_rotation_smoothing` methods and deprecate `reset_smoothing` (my current implementation).

I think the second solution is the best, but there may be other good solutions I have missed.

**Reply by Calinou** [reply]
> Sorry, but I don't understand what an origin shifting is. Do you mean changing position of a camera parent node?

You can see an explanation and a video here: https://www.reddit.com/r/Unity3D/comments/yqy8qt/i_have_created_a_floating_point_origin_system_for/

(It's for Unity, but this concept can be implemented in any engine including Godot. It can be done in both 2D and 3D.)

Godot has [double-precision physics support](https://docs.godotengine.org/en/latest/tutorials/physics/large_world_coordinates.html) since 4.0, so origin shifting isn't needed as often as it used to, but it requires custom builds as no official builds with double precision physics are currently distributed.

**Reply by farfind-gpt** [reply]
Thanks, now I understand.

So, which solution is the best? Or is it better to do nothing until there are more opinions?

**Reply by Calinou** [reply]
> So, which solution is the best? Or is it better to do nothing until there are more opinions?

If you have a working implementation, you can open a PR now (remember to link this discussion).
